Any way to use QuickChick without opam?

Call me old, but I like to compile Coq directly, not via opam. Now, is there any way I can use QuickChick with a version of Coq that is compiled from source with the `-local` configure flag?

I'm currently trying this for Coq 8.7.1, but getting the following error:
```
ocamlbuild -use-ocamlfind -pkg coq.lib -cflags -rectypes -tag thread quickChickTool/quickChickTool.byte
+ ocamlfind ocamldep -package coq.lib -modules quickChickTool/quickChickTool.ml > quickChickTool/quickChickTool.ml.depends
ocamlfind: Package `coq.lib' not found
Command exited with code 2.
Compilation unsuccessful after building 1 target (0 cached) in 00:00:00.
make: *** [Makefile:52: quickChickTool.byte] Error 10
```
I guess it's related to not finding the `META.coq` file or something like that.

--- a/Makefile
+++ b/Makefile
@@ -49,7 +49,7 @@ src/%.cmo : src/%.ml
 quickChickTool: $(QCTOOL_DIR)/$(QCTOOL_EXE)
 
 $(QCTOOL_DIR)/$(QCTOOL_EXE): $(QCTOOL_SRC)
-	cd $(QCTOOL_DIR); ocamlbuild -use-ocamlfind $(QCTOOL_EXE)
+	cd $(QCTOOL_DIR); ocamlbuild -pkg unix -use-ocamlfind $(QCTOOL_EXE)
 
 tests:
 #	$(MAKE) -C examples/ifc-basic test

--- a/quickChickTool/_tags
+++ b/quickChickTool/_tags
@@ -1,1 +1,1 @@
-true: package(coq.lib), rectypes, thread, traverse
+true: rectypes, thread, traverse

--- a/quickChickTool/quickChickTool.ml
+++ b/quickChickTool/quickChickTool.ml
@@ -248,6 +248,19 @@ let gather_all_vs_from_dir fs =
   in loop fs;
   !all_vs
 
+let split_words s =
+  let i = ref 0 in
+  let words = ref [] in
+  let testchar j c =
+    if String.contains " \r\t\n" c then (
+      (if !i < j then words := String.sub s !i (j - !i) :: !words);
+      i := j + 1
+    )
+  in
+  s |> String.iteri testchar;
+  testchar (String.length s) '\n';
+  List.rev !words
+
 (* Find all .v files listed by file fpath. Assumes filenames have no
  * whitespace, and recognizes single-line comments prefixed with '#'. *)
 let vs_from_file fpath =
@@ -260,7 +273,7 @@ let vs_from_file fpath =
                | i -> String.sub line 0 i
     in
     line
-      |> String.split_on_char ' '
+      |> split_words
       |> List.iter (fun w ->
           if Filename.check_suffix w ".v" then
             files := w :: !files)
@@ -469,6 +482,11 @@ let is_dir f =
   try Sys.is_directory f
   with Sys_error _ -> false
 
+let option_map f o =
+  match o with
+  | Some x -> Some (f x)
+  | None -> None
+
 let rec parse_file_or_dir file_name =
   try
     debug "[parse_file_or_dir %s]\n" file_name;
@@ -525,11 +543,11 @@ let rec parse_file_or_dir file_name =
                        (fun sec -> {sec with sec_nodes = collapse sec.sec_nodes}) result in
 
         let fix_extends extends =
-          Str.split (Str.regexp "[ \r\n\t]") (String.concat "" extends) in
+          split_words (String.concat "" extends) in
         let result = List.map
           (fun sec ->
             {sec with sec_extends =
-               Option.map (fun e -> {e with ext_extends = fix_extends e.ext_extends}) sec.sec_extends}
+               option_map (fun e -> {e with ext_extends = fix_extends e.ext_extends}) sec.sec_extends}
           ) result in
         let fixed_default =
           match result with
@@ -557,7 +575,7 @@ let rec section_length_of_fs fs =
 
 (* TODO: "trim" is a confusing name for this function! *)
 let trim s =
-  match Str.split (Str.regexp "[ \r\n\t]") s with
+  match split_words s with
   | [] -> ""
   | h :: _ -> h
 
@@ -716,7 +734,7 @@ let main =
     let all_vs = gather_all_vs (Dir (s, fss)) in
     let extractions = List.map (fun s -> Filename.basename s) all_vs in
     let mk_import s =
-      let splits = List.tl (Str.split (Str.regexp "/") s) in
+      let splits = List.tl (String.split_on_char '/' s) in
       String.concat "." splits in
 
     let imports = List.map (fun s -> (if !top = "" then "" else !top ^ ".") ^ (mk_import s)) all_vs in

--- a/quickChickTool/quickChickTool.ml
+++ b/quickChickTool/quickChickTool.ml
@@ -248,12 +248,24 @@ let gather_all_vs_from_dir fs =
   in loop fs;
   !all_vs
 
-open CoqProject_file
-
-let vs_from_file f =
-  let project = CoqProject_file.read_project_file f in
-  List.map (fun s -> match s with {thing = str; _} -> str)
-    project.v_files
+(* Find all .v files listed by file fpath. Assumes filenames have no
+ * whitespace, and recognizes single-line comments prefixed with '#'. *)
+let vs_from_file fpath =
+  let f = open_in fpath in
+  let files = ref [] in
+  try while true do (* Read the whole file f *)
+    let line = input_line f in
+    let line = match String.index line '#' with
+               | exception Not_found -> line
+               | i -> String.sub line 0 i
+    in
+    line
+      |> String.split_on_char ' '
+      |> List.iter (fun w ->
+          if Filename.check_suffix w ".v" then
+            files := w :: !files)
+  done; assert false with
+  | End_of_file -> !files
 
 let gather_all_vs_from_file f fs =
   let included = vs_from_file f in

--- a/quickChickTool/quickChickToolLexer.mll
+++ b/quickChickTool/quickChickToolLexer.mll
@@ -6,7 +6,7 @@ open QuickChickToolTypes
 (* Function to increase line count in lexbuf *)
 let line_incs s lexbuf =
 (*  Printf.printf "Read: %s\n" s; *)
-  let splits = Str.split_delim (Str.regexp "\n") s in 
+  let splits = String.split_on_char '\n' s in
   let pos = lexbuf.Lexing.lex_curr_p in
 (* Printf.printf "Was in line %d, position %d\n" pos.pos_lnum (pos.pos_cnum - pos.pos_bol); *)
   lexbuf.Lexing.lex_curr_p <- {

